src/main/java/org/thoughtcrime/securesms/qr/QrCodeHandler.java

```diff
@@ -89,6 +89,18 @@ public void handleQrData(String rawString) {
                 BackupTransferActivity.appendSSID(activity, alertDialog.findViewById(android.R.id.message));
                 return;
 
+            case DcContext.DC_QR_PROXY:
+                builder.setTitle(R.string.proxy_use_proxy);
+                builder.setMessage(activity.getString(R.string.proxy_use_proxy_confirm, qrParsed.getText1()));
+                builder.setPositiveButton(R.string.proxy_use_proxy, (dlg, btn) -> {
+                    dcContext.setConfigFromQr(rawString);
+                    dcContext.stopIo();
+                    dcContext.startIo();
```

Comment:
this sets implicitly the default proxy and enables proxy settings, right?

and for that to take place, stop/start IO is needed?

Reply:
yes, it sets the configs but current IMAP connection is not reconnected via the proxy automatically, so restarting IO is necessary